pkg/cosign/verify.go

```diff
@@ -83,20 +83,31 @@ func LoadPublicKey(keyRef string) (*ecdsa.PublicKey, error) {
 	return ed, nil
 }
 
-func VerifySignature(pubkey *ecdsa.PublicKey, base64sig string, payload []byte) error {
-	signature, err := base64.StdEncoding.DecodeString(base64sig)
-	if err != nil {
-		return err
-	}
+type Verifier interface {
+	Verify(ctx context.Context, payload, signature []byte) error
+}
+
+type ECDSAVerifier struct {
+	PubKey *ecdsa.PublicKey
+}
 
+func (v *ECDSAVerifier) Verify(_ context.Context, payload, signature []byte) error {
```

Comment:
maybe just drop the unused context here since we dont even pass it below?

Reply:
`syntax error: mixed named and unnamed function parameters`